### PR TITLE
Visual Studio TestExplorer detects the google tests

### PR DESCRIPTION
**Pull request by twollgam**
### What does this PR do?

This PR adds files so that the Visual Studio TestExplorer  detects the google tests.

The anonymous namespaces will be removed. Now the TestExplorer shows the tests under <Empty Namespace>.  Else it is mixed with "anonymous namespace".

image

### Closes Issue(s)

None

### Motivation

The TestExplorer makes it easy to run single tests and debug them.

### Testing

- Only developers on windows with visual studio. 
- TestExplorer must be installed and (included with vs 2022) Google Test Adapter must be installed.
- Open TestExplorer
- Compile whole project
- Tests should be displayed

### CODE REVIEW

**Comment by twollgam** [comment]
@Baklap4 would be nice if you can test this one , review and merge it. There is no productive code involved. 

Cause it is very helpful while developing, I hope you can do it soon.

**Review by Baklap4 — CHANGES_REQUESTED** [review]
Not sure the testenvironments.json is needed since it doesn't contain info.

As for the rest it looks good thanks that you've enabled the test explorer! 🎉

File: testEnvironments.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿{
```
[the remark this answers is not in the document]
Comment (twollgam) [reply testEnvironments.json]: Don't where it comes from, but it is not needed 👍 Removed it ...

**Comment by Baklap4** [comment]
Let's await #981 as i think i found the reason why macos build is failing....

**Review by Baklap4 — APPROVED** [review]

**Comment by Baklap4** [comment]
Can be rebased so that it can be merged, maybe still await 981 before rebasing so you dont have to build twice

**Pull request merged**